src/k2/transport/Payload.h

```diff
@@ -531,16 +532,32 @@ class Payload {
         return size + sizeof(size);
     }
 
-    // for type: std::decimal::decimal64
+    // for type: boost::multiprecision::cpp_dec_float_50
     template <typename T>
-    std::enable_if_t<std::is_same_v<T, std::decimal::decimal64>, size_t> getSerializedSizeOf() {
-        return sizeof(std::decimal::decimal64::__decfloat64);
+    std::enable_if_t<std::is_same_v<T, boost::multiprecision::cpp_dec_float_50>, size_t> getSerializedSizeOf() {
```

Comment:
you can now combine these two methods since the api is the same for both dec50 and dec100. Use the template switch to check for either dev50 || dec100